### PR TITLE
Mark notable breaking changes for 7.0

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds a tag for each breaking change that could be considered notable.

### CODE REVIEW

**Comment by jasontedor** [comment]
@eskibars @jpountz This is my take on what is considered notable breaking changes. I am happy to hear your thoughts as well. I did try to keep this list not long so if we do want to add more to it, I think we should also consider taking away some of the items that I have already marked as notable, otherwise it will be laundry list.

File: docs/reference/migration/migrate_7_0/settings.asciidoc [hunk 1]
```diff
@@ -43,11 +45,13 @@ of the node id. It can still be configured explicitly in `elasticsearch.yml`.
 
 [float]
 [[remove-http-enabled]]
+//tag::notable-breaking-changes[]
```
Comment (jpountz) [inline docs/reference/migration/migrate_7_0/settings.asciidoc]: I'm wondering how many users are disabling HTTP, should we keep it a non-notable breaking change?

File: docs/reference/migration/migrate_7_0/settings.asciidoc [hunk 2]
```diff
@@ -59,6 +63,7 @@ provided little value. The setting `http.pipelining.max_events` can still be use
 limit the number of pipelined requests in-flight.
 
 [float]
+//tag::notable-breaking-changes[]
```
Comment (jpountz) [inline docs/reference/migration/migrate_7_0/settings.asciidoc]: This one won't fail any upgrade if my understanding is correct, so I'd be leaning towards removing it.

File: docs/reference/migration/migrate_7_0/logging.asciidoc [hunk 3]
```diff
@@ -3,6 +3,7 @@
 === Logging changes
 
 [float]
+//tag::notable-breaking-changes[]
```
Comment (jpountz) [inline docs/reference/migration/migrate_7_0/logging.asciidoc]: I'd maybe put these changes around JSON log files in the notable highlights but not in the notable breaking changes?

**Review by jpountz — COMMENTED** [review]
I left comments around whether some breaking changes are actually notable but in general this looks good to me. I reviewed the list of breaking changes that we document and didn't find one that I would put on the list that you didn't.

**Review by eskibars — CHANGES_REQUESTED** [review]
I left a few comments.  It looks like there is one close-tag without an open.  There are 2 others that I would consider adding:
1) search.max_buckets: Based on the number of conversations I've had with users that haven't configured this and had bad things happen, I suspect we'll want to make this prominent.
2) The security realm type prefix requirements.  We've got the non-prefixed format in a lot of docs/blogs/examples, even recent ones (e.g. https://www.elastic.co/guide/en/elasticsearch/reference/6.7/security-settings.html#realm-settings) and this could be very surprising and frustrating if users aren't aware of it.

**Comment by lcawl** [comment]
You can preview this content in the Installation and Upgrade Guide here: 
https://doc-review-63bf6.firebaseapp.com/x4/elastic-stack-breaking-changes.html#elasticsearch-breaking-changes

File: docs/reference/migration/migrate_7_0/search.asciidoc [hunk 4]
```diff
@@ -205,6 +205,7 @@ on whether queries need to access score or not. As a result `bool` queries with
 `minimum_should_match` to 1. This behavior has been deprecated in the previous
 major version.
 
+//tag::notable-breaking-changes[]
```
Comment (lcawl) [inline docs/reference/migration/migrate_7_0/search.asciidoc]: There's a <1> in the example in this section that isn't explained.

**Comment by jasontedor** [comment]
@eskibars I've responded to your feedback. I added the two that you suggested, and removed the documentation distribution change.

**Review by eskibars — APPROVED** [review]
Thanks @jasontedor .  This LGTM

**Pull request merged**